common/test/unit/com/thoughtworks/go/config/materials/git/GitMaterialTest.java

```diff
@@ -237,6 +237,19 @@ public void shouldNotDeleteAndRecheckoutDirectoryWhenUrlsOnlyDifferInFileProtoco
     }
 
     @Test
+    public void shouldNotDeleteAndRecheckoutDirectoryWhenBranchIsBlank() throws Exception {
+        git.latestModification(workingDir, new TestSubprocessExecutionContext());
+
+        File shouldNotBeRemoved = new File(new File(workingDir, ".git"), "shouldNotBeRemoved");
+        FileUtils.writeStringToFile(shouldNotBeRemoved, "gundi");
+        assertThat(shouldNotBeRemoved.exists(), is(true));
+
+        git = new GitMaterial(repositoryUrl, " ");
+        git.latestModification(workingDir, new TestSubprocessExecutionContext());
+        assertThat("Should not have deleted whole folder", shouldNotBeRemoved.exists(), is(true));
+    }
+
```

Comment:
Don't think we need this test.
Rather we need one for ```isRepositoryChanged``` to check for empty strings.